### ISSUE TITLE
emqx-edge bridge can't be found from the inflight error

### ISSUE THREAD

**Issue by moluzhui** [emqx/emqx#3629]
**Please describe your problem in detail, if necessary, you can upload the log file through the attachment**:
I use the `EMQ X Edge:v4.1.1 container`  to bridge the `EMQ X Broker:v4.1.1` container. 

I implemented bridging subscriptions to `multiple topics` by modifying emqx_bridge_mqtt.conf, as follows

### the modification of emqx_bridge_mqtt.conf
```
## Forward messages to the mountpoint of an AWS IoTHUB
##
## Value: String
bridge.mqtt.aws.forward_mountpoint = /

## Need to subscribe to AWS topics
##
## Value: String
bridge.mqtt.aws.subscription.1.topic = data/2mNZ3VEVivxfHsqZp7Xx8V/#

## Need to subscribe to AWS topics QoS.
##
## Value: Number
bridge.mqtt.aws.subscription.1.qos = 2


bridge.mqtt.aws.subscription.2.topic = cmd/2mNZ3VEVivxfHsqZp7Xx8V/#

bridge.mqtt.aws.subscription.2.qos = 2

## A mountpoint that receives messages from AWS IoTHUB
##
## Value: String
## bridge.mqtt.aws.receive_mountpoint = receive/aws/
```

And start the EMQ X Edge bridge service using the following configuration and commands
### env.list
```
EMQX_BRIDGE__MQTT__AWS__ADDRESS=xx.xx.xxx.xx:1883
EMQX_BRIDGE__MQTT__AWS__START_TYPE=auto
EMQX_BRIDGE__MQTT__AWS__FORWARDS=cmd/gFajLFAb8fVfAW4p55r6sM/#,data/gFajLFAb8fVfAW4p55r6sM/#
EMQX_BRIDGE__MQTT__AWS__FORWARD_MOUNTPOINT=/
EMQX_BRIDGE__MQTT__AWS__SUBSCRIPTION__1__TOPIC=data/2mNZ3VEVivxfHsqZp7Xx8V/#
EMQX_BRIDGE__MQTT__AWS__SUBSCRIPTION__2__TOPIC=cmd/2mNZ3VEVivxfHsqZp7Xx8V/#
EMQX_BRIDGE__MQTT__AWS__SUBSCRIPTION__1__QOS=2
EMQX_BRIDGE__MQTT__AWS__SUBSCRIPTION__2__QOS=2
EMQX_BRIDGE__MQTT__AWS__CLEAN_START=false
```

### docker command
```
docker run -d --name emqxedge411 -p 1883:1883 --env-file env.list emqx/emqx-edge:v4.1.1
```

There were no problems during the two days of testing, but an Inflight error occurred today, as follows
```
EMQ X Edge 4.1.1 is running now!
Eshell V10.7.1  (abort with ^G)
(6ddc3c74620e@172.17.0.2)1> ['2020-07-27T02:13:14Z']:emqx start

===== ALIVE Mon Jul 27 02:28:07 UTC 2020

===== ALIVE Mon Jul 27 02:43:07 UTC 2020

===== ALIVE Mon Jul 27 02:58:07 UTC 2020

===== ALIVE Mon Jul 27 03:13:07 UTC 2020

===== ALIVE Mon Jul 27 03:28:07 UTC 2020

===== ALIVE Mon Jul 27 03:43:07 UTC 2020

===== ALIVE Mon Jul 27 03:58:07 UTC 2020

===== ALIVE Mon Jul 27 04:13:07 UTC 2020

===== ALIVE Mon Jul 27 04:28:07 UTC 2020

===== ALIVE Mon Jul 27 04:43:07 UTC 2020

===== ALIVE Mon Jul 27 04:58:07 UTC 2020
2020-07-27 05:00:10.550 [error] emqtt(bridge_aws): The connection error occured tcp_error, reason:etimedout
(6ddc3c74620e@172.17.0.2)1>
===== ALIVE Mon Jul 27 05:15:10 UTC 2020

===== ALIVE Mon Jul 27 05:30:10 UTC 2020

===== ALIVE Mon Jul 27 05:45:10 UTC 2020

===== ALIVE Mon Jul 27 06:00:10 UTC 2020

===== ALIVE Mon Jul 27 06:15:10 UTC 2020

===== ALIVE Mon Jul 27 06:30:10 UTC 2020

===== ALIVE Mon Jul 27 06:45:11 UTC 2020

===== ALIVE Mon Jul 27 07:00:11 UTC 2020
2020-07-27 07:04:43.706 [error] emqtt(bridge_aws): The connection error occured tcp_error, reason:etimedout
(6ddc3c74620e@172.17.0.2)1>
===== ALIVE Mon Jul 27 07:19:43 UTC 2020

===== ALIVE Mon Jul 27 07:34:43 UTC 2020

===== ALIVE Mon Jul 27 07:49:43 UTC 2020

===== ALIVE Mon Jul 27 08:04:43 UTC 2020

===== ALIVE Mon Jul 27 08:19:43 UTC 2020

===== ALIVE Mon Jul 27 08:34:44 UTC 2020

===== ALIVE Mon Jul 27 08:49:44 UTC 2020
2020-07-27 08:54:38.098 [error] [Bridge] Can't be found from the inflight:4
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.354 [error] [Bridge] Can't be found from the inflight:5
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.614 [error] [Bridge] Can't be found from the inflight:6
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.616 [error] [Bridge] Can't be found from the inflight:7
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.618 [error] [Bridge] Can't be found from the inflight:8
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.643 [error] [Bridge] Can't be found from the inflight:9
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.643 [error] [Bridge] Can't be found from the inflight:10
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.644 [error] [Bridge] Can't be found from the inflight:11
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.645 [error] [Bridge] Can't be found from the inflight:12
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.645 [error] [Bridge] Can't be found from the inflight:13
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.646 [error] [Bridge] Can't be found from the inflight:14
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:54:38.669 [error] [Bridge] Can't be found from the inflight:15
...
...
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:55:39.666 [error] [Bridge] Can't be found from the inflight:667
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:55:39.666 [error] [Bridge] Can't be found from the inflight:668
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:55:39.667 [error] [Bridge] Can't be found from the inflight:669
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:55:39.667 [error] [Bridge] Can't be found from the inflight:670
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:55:39.668 [error] [Bridge] Can't be found from the inflight:671
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:55:39.668 [error] [Bridge] Can't be found from the inflight:672
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:58:29.973 [error] [Bridge Connect] Failed to connect with module=emqx_bridge_mqtt
config=[{username,"user"},
        {subscriptions,[{<<"ECCI/2mNZ3VEVivxfHsqZp7Xx8V/#">>,2},
                        {<<"radio/ECCI/2mNZ3VEVivxfHsqZp7Xx8V/#">>,2}]},
        {start_type,auto},
        {ssl_opts,[{versions,['tlsv1.2','tlsv1.1',tlsv1]},
                   {keyfile,"etc/certs/client-key.pem"},
                   {ciphers,["ECDHE-ECDSA-AES256-GCM-SHA384",
                             "ECDHE-RSA-AES256-GCM-SHA384",
                             "ECDHE-ECDSA-AES256-SHA384",
                             "ECDHE-RSA-AES256-SHA384",
                             "ECDHE-ECDSA-DES-CBC3-SHA",
                             "ECDH-ECDSA-AES256-GCM-SHA384",
                             "ECDH-RSA-AES256-GCM-SHA384",
                             "ECDH-ECDSA-AES256-SHA384",
                             "ECDH-RSA-AES256-SHA384",
                             "DHE-DSS-AES256-GCM-SHA384",
                             "DHE-DSS-AES256-SHA256","AES256-GCM-SHA384",
                             "AES256-SHA256","ECDHE-ECDSA-AES128-GCM-SHA256",
                             "ECDHE-RSA-AES128-GCM-SHA256",
                             "ECDHE-ECDSA-AES128-SHA256",
                             "ECDHE-RSA-AES128-SHA256",
                             "ECDH-ECDSA-AES128-GCM-SHA256",
                             "ECDH-RSA-AES128-GCM-SHA256",
                             "ECDH-ECDSA-AES128-SHA256",
                             "ECDH-RSA-AES128-SHA256",
                             "DHE-DSS-AES128-GCM-SHA256",
                             "DHE-DSS-AES128-SHA256","AES128-GCM-SHA256",
                             "AES128-SHA256","ECDHE-ECDSA-AES256-SHA",
                             "ECDHE-RSA-AES256-SHA","DHE-DSS-AES256-SHA",
                             "ECDH-ECDSA-AES256-SHA","ECDH-RSA-AES256-SHA",
                             "AES256-SHA","ECDHE-ECDSA-AES128-SHA",
                             "ECDHE-RSA-AES128-SHA","DHE-DSS-AES128-SHA",
                             "ECDH-ECDSA-AES128-SHA","ECDH-RSA-AES128-SHA",
                             "AES128-SHA"]},
                   {certfile,"etc/certs/client-cert.pem"},
                   {cacertfile,"etc/certs/cacert.pem"}]},
        {ssl,false},
        {retry_interval,20000},
        {proto_ver,v4},
        {password,'***'},
        {max_inflight_size,32},
        {keepalive,60},
        {forward_mountpoint,"/"},
        {clientid,"bridge_aws"},
        {clean_start,false},
        {bridge_mode,true},
        {batch_size,32},
        {address,"xx.xxx.xxx.xx:1883"}]
reason:{error,ack_timeout}
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 08:59:09.653 [error] [Bridge] Can't be found from the inflight:4
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 09:02:00.107 [error] [Bridge Connect] Failed to connect with module=emqx_bridge_mqtt
config=[{username,"user"},
        {subscriptions,[{<<"ECCI/2mNZ3VEVivxfHsqZp7Xx8V/#">>,2},
                        {<<"radio/ECCI/2mNZ3VEVivxfHsqZp7Xx8V/#">>,2}]},
        {start_type,auto},
        {ssl_opts,[{versions,['tlsv1.2','tlsv1.1',tlsv1]},
                   {keyfile,"etc/certs/client-key.pem"},
                   {ciphers,["ECDHE-ECDSA-AES256-GCM-SHA384",
                             "ECDHE-RSA-AES256-GCM-SHA384",
                             "ECDHE-ECDSA-AES256-SHA384",
                             "ECDHE-RSA-AES256-SHA384",
                             "ECDHE-ECDSA-DES-CBC3-SHA",
                             "ECDH-ECDSA-AES256-GCM-SHA384",
                             "ECDH-RSA-AES256-GCM-SHA384",
                             "ECDH-ECDSA-AES256-SHA384",
                             "ECDH-RSA-AES256-SHA384",
                             "DHE-DSS-AES256-GCM-SHA384",
                             "DHE-DSS-AES256-SHA256","AES256-GCM-SHA384",
                             "AES256-SHA256","ECDHE-ECDSA-AES128-GCM-SHA256",
                             "ECDHE-RSA-AES128-GCM-SHA256",
                             "ECDHE-ECDSA-AES128-SHA256",
                             "ECDHE-RSA-AES128-SHA256",
                             "ECDH-ECDSA-AES128-GCM-SHA256",
                             "ECDH-RSA-AES128-GCM-SHA256",
                             "ECDH-ECDSA-AES128-SHA256",
                             "ECDH-RSA-AES128-SHA256",
                             "DHE-DSS-AES128-GCM-SHA256",
                             "DHE-DSS-AES128-SHA256","AES128-GCM-SHA256",
                             "AES128-SHA256","ECDHE-ECDSA-AES256-SHA",
                             "ECDHE-RSA-AES256-SHA","DHE-DSS-AES256-SHA",
                             "ECDH-ECDSA-AES256-SHA","ECDH-RSA-AES256-SHA",
                             "AES256-SHA","ECDHE-ECDSA-AES128-SHA",
                             "ECDHE-RSA-AES128-SHA","DHE-DSS-AES128-SHA",
                             "ECDH-ECDSA-AES128-SHA","ECDH-RSA-AES128-SHA",
                             "AES128-SHA"]},
                   {certfile,"etc/certs/client-cert.pem"},
                   {cacertfile,"etc/certs/cacert.pem"}]},
        {ssl,false},
        {retry_interval,20000},
        {proto_ver,v4},
        {password,'***'},
        {max_inflight_size,32},
        {keepalive,60},
        {forward_mountpoint,"/"},
        {clientid,"bridge_aws"},
        {clean_start,false},
        {bridge_mode,true},
        {batch_size,32},
        {address,"xx.xxx.xxx.xx:1883"}]
reason:{{shutdown,tcp_closed},{tcp_closed,#Port<0.4949>}}
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 09:02:45.167 [error] [Bridge Connect] Failed to connect with module=emqx_bridge_mqtt
config=[{username,"user"},
        {subscriptions,[{<<"ECCI/2mNZ3VEVivxfHsqZp7Xx8V/#">>,2},
                        {<<"radio/ECCI/2mNZ3VEVivxfHsqZp7Xx8V/#">>,2}]},
        {start_type,auto},
        {ssl_opts,[{versions,['tlsv1.2','tlsv1.1',tlsv1]},
                   {keyfile,"etc/certs/client-key.pem"},
                   {ciphers,["ECDHE-ECDSA-AES256-GCM-SHA384",
                             "ECDHE-RSA-AES256-GCM-SHA384",
                             "ECDHE-ECDSA-AES256-SHA384",
                             "ECDHE-RSA-AES256-SHA384",
                             "ECDHE-ECDSA-DES-CBC3-SHA",
                             "ECDH-ECDSA-AES256-GCM-SHA384",
                             "ECDH-RSA-AES256-GCM-SHA384",
                             "ECDH-ECDSA-AES256-SHA384",
                             "ECDH-RSA-AES256-SHA384",
                             "DHE-DSS-AES256-GCM-SHA384",
                             "DHE-DSS-AES256-SHA256","AES256-GCM-SHA384",
                             "AES256-SHA256","ECDHE-ECDSA-AES128-GCM-SHA256",
                             "ECDHE-RSA-AES128-GCM-SHA256",
                             "ECDHE-ECDSA-AES128-SHA256",
                             "ECDHE-RSA-AES128-SHA256",
                             "ECDH-ECDSA-AES128-GCM-SHA256",
                             "ECDH-RSA-AES128-GCM-SHA256",
                             "ECDH-ECDSA-AES128-SHA256",
                             "ECDH-RSA-AES128-SHA256",
                             "DHE-DSS-AES128-GCM-SHA256",
                             "DHE-DSS-AES128-SHA256","AES128-GCM-SHA256",
                             "AES128-SHA256","ECDHE-ECDSA-AES256-SHA",
                             "ECDHE-RSA-AES256-SHA","DHE-DSS-AES256-SHA",
                             "ECDH-ECDSA-AES256-SHA","ECDH-RSA-AES256-SHA",
                             "AES256-SHA","ECDHE-ECDSA-AES128-SHA",
                             "ECDHE-RSA-AES128-SHA","DHE-DSS-AES128-SHA",
                             "ECDH-ECDSA-AES128-SHA","ECDH-RSA-AES128-SHA",
                             "AES128-SHA"]},
                   {certfile,"etc/certs/client-cert.pem"},
                   {cacertfile,"etc/certs/cacert.pem"}]},
        {ssl,false},
        {retry_interval,20000},
        {proto_ver,v4},
        {password,'***'},
        {max_inflight_size,32},
        {keepalive,60},
        {forward_mountpoint,"/"},
        {clientid,"bridge_aws"},
        {clean_start,false},
        {bridge_mode,true},
        {batch_size,32},
        {address,"xx.xxx.xxx.xx:1883"}]
reason:{{shutdown,tcp_closed},{tcp_closed,#Port<0.4959>}}
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 09:03:43.318 [error] [Bridge] Can't be found from the inflight:10
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 09:03:43.319 [error] [Bridge] Can't be found from the inflight:11
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 09:03:45.014 [error] [Bridge] Can't be found from the inflight:12
(6ddc3c74620e@172.17.0.2)1> 2020-07-27 09:03:45.015 [error] [Bridge] Can't be found from the inflight:13
...
...
```

This situation causes the bridge message to fail to be sent, but after some time it is automatically sent, although `the order of sending` appears to have been changed.

What causes an **Inflight** error?

Looking forward to your answers

**Comment by HJianBo** [comment]
@moluzhui Thanks for your feedback. We try to fix it at v4.2.0

see: https://github.com/emqx/emqx-bridge-mqtt/issues/81

**Comment by moluzhui** [comment]
> @moluzhui Thanks for your feedback. We try to fix it at v4.2.0
> 
> see: [emqx/emqx-bridge-mqtt#81](https://github.com/emqx/emqx-bridge-mqtt/issues/81)

@HJianBo  I actually had a `similar problem` with `another bridge test`. This may also be an issue that needs to be addressed, related to the `large_heap warning`.

the configuration of emqx-edge and docker command is as follow:
```
EMQX_BRIDGE__MQTT__AWS__ADDRESS=xx.xxx.xxx.xxx:1883
EMQX_BRIDGE__MQTT__AWS__START_TYPE=auto
EMQX_BRIDGE__MQTT__AWS__FORWARDS=data/321/#
EMQX_BRIDGE__MQTT__AWS__FORWARD_MOUNTPOINT=/
EMQX_BRIDGE__MQTT__AWS__SUBSCRIPTION__1__TOPIC=cmd/123/#
EMQX_BRIDGE__MQTT__AWS__SUBSCRIPTION__1__QOS=2
EMQX_BRIDGE__MQTT__AWS__CLEAN_START=false

docker run -d -p 1883:1883 --env-file env.list --name edgeemqx411 emqx/emqx-edge:v4.1.1
```

In the EMQX-Edge bridged to EMQ X-Broker, an application has `four containers`, three of which are connected to EMQX-Edge, and they communicate `frequently with each other`, and some containers send bridging data to EMQX-Broker in the expectation that it will be received and stored in a timely manner by application components that subscribe to the topic on the cloud.

The three container logs that subscribe to EMQX-Edge show that they can send and receive messages in a timely manner, but the bridged data **cannot be forwarded in a timely manner**.

After waiting for some time, EMQx-Edge logs appear with the **lager_heap warning**, which appears to cause the bridge to abort, then EMQX stops, with an **Inflight error**, and then the bridged forward messages are `received` by the client that is connected to emQX-Broker and appear to be **unstopped and out of order**. The bridged clients are found to be constantly disconnected and connected through the `emqx_ctl clients list`.

At this time, it is observed through `htop` that the **memory** of EMQX-Edge `rises rapidly for a period of time and then drops`.

Part of the EMQX-Edge log is shown below
```
EMQ X Edge 4.1.1 is running now!
Eshell V10.7.1  (abort with ^G)
(a96d8f209f57@172.17.0.2)1> ['2020-07-28T12:09:36Z']:emqx start
2020-07-28 12:14:07.581 [warning] [SYSMON] large_heap warning: pid = <0.1332.0>, info: [{old_heap_block_size,6189440},
                                             {heap_block_size,2984878},
                                             {mbuf_size,0},
                                             {stack_size,381},
                                             {old_heap_size,2860338},
                                             {heap_size,963561}]
[{initial_call,{proc_lib,init_p,5}},
 {current_function,{erlang,integer_to_list,1}},
 {registered_name,[]},
 {status,running},
 {message_queue_len,3},
 {group_leader,<0.1324.0>},
 {priority,normal},
 {trap_exit,true},
 {reductions,3351915},
 {last_calls,false},
 {catchlevel,5},
 {trace,0},
 {suspending,[]},
 {sequential_trace_token,[]},
 {error_handler,error_handler},
 {memory,73395776},
 {total_heap_size,9174318},
 {heap_size,2984878},
 {stack_size,376},
 {min_heap_size,233}]
(a96d8f209f57@172.17.0.2)1> 2020-07-28 12:14:07.334 [error] ** State machine <0.1332.0> terminating
** Last event = {{call,{<0.1327.0>,#Ref<0.1970789639.1287913473.1579>}},
                 {publish,{mqtt_msg,1,false,false,undefined,
                                    <<"/data/321/app/app/sensor1/targetcontainer">>,
                                    undefined,
                                    <<128,3,125,113,0,40,88,4,0,0,0,116,121,
                                      112,101,113,1,88,4,0,0,0,100,97,116,
                                      97,113,2,88,8,0,0,0,99,111,110,116,
                                     ......
                                     ......
                                     113,19,116,113,20,98,75,1,71,65,215,
                                      200,6,84,192,156,136,101,117,46>>}}}
** When server state  = {connected,
                         {state,undefined,<0.1327.0>,
                          #{disconnected => #Fun<emqx_bridge_mqtt.4.42003753>,
                            puback => #Fun<emqx_bridge_mqtt.2.42003753>,
                            publish => #Fun<emqx_bridge_mqtt.3.42003753>},
                          "xx.xxx.xxx.xxx",1883,[],emqtt_sock,#Port<0.19>,[],
                          60000,true,<<"bridge_aws">>,false,<<"user">>,
                          <<"passwd">>,4,<<"MQTT">>,60,
                          #Ref<0.644288161.752615425.148960>,true,false,false,
                          {mqtt_msg,0,false,false,undefined,undefined,
                           undefined,undefined},
                          #{},[],
                          #{<<"cmd/123/#">> =>
                             #{nl => 0,qos => 2,rap => 0,rh => 0}},
                          infinity,
                          #{253 =>
                             {publish,
                              {mqtt_msg,1,false,false,253,
                               <<"/data/321/app/app/sensor1/targetcontainer">>,
                               undefined,
                               <<128,3,125,113,0,40,88,4,0,0,0,116,121,112,
                                 101,113,1,88,4,0,0,0,100,97,116,97,113,2,88,
                                 8,0,0,0,99,111,110,116,101,110,116,115,113,3,
                                 93,113,4,40,71,64,104,240,0,0,0,0,0,88,6,0,0,
                                     ......
                                     ......
                                 20,98,75,1,71,65,215,200,6,83,84,236,228,101,
                                 117,46>>},
                              {1595,939149,379144}},
                            254 =>
                             {publish,
                              {mqtt_msg,1,false,false,254,
                               <<"/data/321/app/app/sensor1/targetcontainer">>,
                               undefined,
                               <<128,3,125,113,0,40,88,4,0,0,0,116,121,112,
                                 101,113,1,88,4,0,0,0,100,97,116,97,113,2,88,
                                     ......
                                     ......
                                 113,20,98,75,1,71,65,215,200,6,83,97,45,49,
                                 101,117,46>>},
                              {1595,939149,560077}},
                            256 =>
                             {publish,
                              {mqtt_msg,1,false,false,256,
                               <<"/data/321/app/app/sensor1/targetcontainer">>,
                               undefined,
                               <<128,3,125,113,0,40,88,4,0,0,0,116,121,112,
                                 101,113,1,88,4,0,0,0,100,97,116,97,113,2,88,
                                     ......
                                     ......
71,65,215,200,6,83,186,109,203,101,117,46>>},
                              {1595,939150,966537}},
                            257 =>
                             {publish,
                              {mqtt_msg,1,false,false,257,
                               <<"/data/321/app/app/sensor1/targetcontainer">>,
                               undefined,
                               <<128,3,125,113,0,40,88,4,0,0,0,116,121,112,
                                 101,113,1,88,4,0,0,0,100,97,116,97,113,2,88,
                                     ......
                                     ......
                                 87,83,82,87,82,81,146,152,142,146,157,161,
                                 138,144,148,143,138,141,129,121,122,100,99,eheap_alloc: Cannot allocate 4723591984 bytes of memory (
                                 101,167,180,184,155,157,164,180,172,181,53,
                                     ......
                                     ......
                                 134,128,125,126,125,122,123,87,84,85,67,68,
                                 68,83,91,90,143,155,154,147,161,160,147,161,of type "heap").
heart: Tue Jul 28 12:28:47 2020: Erlang is crashing .. (waiting for crash dump file)
[os_mon] cpu supervisor port (cpu_sup): Erlang has closed
[os_mon] memory supervisor port (memsup): Erlang has closed
['2020-07-28T12:28:48Z']:emqx not running, waiting for recovery in 20 seconds

=====
===== LOGGING STARTED Tue Jul 28 12:28:49 UTC 2020
=====
Exec: /opt/emqx/erts-10.7.1/bin/erlexec -boot /opt/emqx/releases/v4.1.1/emqx -mode embedded -boot_var ERTS_LIB_DIR /opt/emqx/erts-10.7.1/../lib -mnesia dir "/opt/emqx/data/mnesia/7ef2c6278889@172.17.0.2" -config /opt/emqx/data/configs/app.2020.07.28.12.28.50.config -args_file /opt/emqx/data/configs/vm.2020.07.28.12.28.50.args -vm_args /opt/emqx/data/configs/vm.2020.07.28.12.28.50.args -start_epmd false -epmd_module ekka_epmd -proto_dist ekka -- console
Root: /opt/emqx
/opt/emqx
['2020-07-28T12:28:53Z']:emqx not running, waiting for recovery in 15 seconds
heart_beat_kill_pid = 2796
Starting emqx on node 7ef2c6278889@172.17.0.2
Start http:management listener on 8081 successfully.
Start http:dashboard listener on 18083 successfully.
Start mqtt:tcp listener on 127.0.0.1:11883 successfully.
Start mqtt:tcp listener on 0.0.0.0:1883 successfully.
Start mqtt:ws listener on 0.0.0.0:8083 successfully.
Start mqtt:ssl listener on 0.0.0.0:8883 successfully.
Start mqtt:wss listener on 0.0.0.0:8084 successfully.
EMQ X Edge 4.1.1 is running now!
Eshell V10.7.1  (abort with ^G)
(7ef2c6278889@172.17.0.2)1> 2020-07-28 12:29:08.988 [error] [Bridge Connect] Failed to connect with module=emqx_bridge_mqtt
config=[{username,"user"},
        {subscriptions,[{<<"cmd/123/#">>,2}]},
        {start_type,auto},
        {ssl_opts,[{versions,['tlsv1.2','tlsv1.1',tlsv1]},
                   {keyfile,"etc/certs/client-key.pem"},
                   {ciphers,["ECDHE-ECDSA-AES256-GCM-SHA384",
                             "ECDHE-RSA-AES256-GCM-SHA384",
                             "ECDHE-ECDSA-AES256-SHA384",
                             "ECDHE-RSA-AES256-SHA384",
                             "ECDHE-ECDSA-DES-CBC3-SHA",
                             "ECDH-ECDSA-AES256-GCM-SHA384",
                             "ECDH-RSA-AES256-GCM-SHA384",
                             "ECDH-ECDSA-AES256-SHA384",
                             "ECDH-RSA-AES256-SHA384",
                             "DHE-DSS-AES256-GCM-SHA384",
                             "DHE-DSS-AES256-SHA256","AES256-GCM-SHA384",
                             "AES256-SHA256","ECDHE-ECDSA-AES128-GCM-SHA256",
                             "ECDHE-RSA-AES128-GCM-SHA256",
                             "ECDHE-ECDSA-AES128-SHA256",
                             "ECDHE-RSA-AES128-SHA256",
                             "ECDH-ECDSA-AES128-GCM-SHA256",
                             "ECDH-RSA-AES128-GCM-SHA256",
                             "ECDH-ECDSA-AES128-SHA256",
                             "ECDH-RSA-AES128-SHA256",
                             "DHE-DSS-AES128-GCM-SHA256",
                             "DHE-DSS-AES128-SHA256","AES128-GCM-SHA256",
                             "AES128-SHA256","ECDHE-ECDSA-AES256-SHA",
                             "ECDHE-RSA-AES256-SHA","DHE-DSS-AES256-SHA",
                             "ECDH-ECDSA-AES256-SHA","ECDH-RSA-AES256-SHA",
                             "AES256-SHA","ECDHE-ECDSA-AES128-SHA",
                             "ECDHE-RSA-AES128-SHA","DHE-DSS-AES128-SHA",
                             "ECDH-ECDSA-AES128-SHA","ECDH-RSA-AES128-SHA",
                             "AES128-SHA"]},
                   {certfile,"etc/certs/client-cert.pem"},
                   {cacertfile,"etc/certs/cacert.pem"}]},
        {ssl,false},
        {retry_interval,20000},
        {proto_ver,v4},
        {password,'***'},
        {max_inflight_size,32},
        {keepalive,60},
        {forward_mountpoint,"/"},
        {clientid,"bridge_aws"},
        {clean_start,false},
        {bridge_mode,true},
        {batch_size,32},
        {address,"xx.xxx.xxx.xxx:1883"}]
reason:{{shutdown,tcp_closed},{tcp_closed,#Port<0.18>}}
(7ef2c6278889@172.17.0.2)1>
===== Tue Jul 28 12:37:55 UTC 2020
2020-07-28 12:37:55.363 [error] [Bridge Connect] Failed to connect with module=emqx_bridge_mqtt
config=[{username,"user"},
        {subscriptions,[{<<"cmd/123/#">>,2}]},
        {start_type,auto},
        {ssl_opts,[{versions,['tlsv1.2','tlsv1.1',tlsv1]},
                   {keyfile,"etc/certs/client-key.pem"},
                   {ciphers,["ECDHE-ECDSA-AES256-GCM-SHA384",
                             "ECDHE-RSA-AES256-GCM-SHA384",
                             "ECDHE-ECDSA-AES256-SHA384",
                             "ECDHE-RSA-AES256-SHA384",
                             "ECDHE-ECDSA-DES-CBC3-SHA",
                             "ECDH-ECDSA-AES256-GCM-SHA384",
                             "ECDH-RSA-AES256-GCM-SHA384",
                             "ECDH-ECDSA-AES256-SHA384",
                             "ECDH-RSA-AES256-SHA384",
                             "DHE-DSS-AES256-GCM-SHA384",
                             "DHE-DSS-AES256-SHA256","AES256-GCM-SHA384",
                             "AES256-SHA256","ECDHE-ECDSA-AES128-GCM-SHA256",
                             "ECDHE-RSA-AES128-GCM-SHA256",
                             "ECDHE-ECDSA-AES128-SHA256",
                             "ECDHE-RSA-AES128-SHA256",
                             "ECDH-ECDSA-AES128-GCM-SHA256",
                             "ECDH-RSA-AES128-GCM-SHA256",
                             "ECDH-ECDSA-AES128-SHA256",
                             "ECDH-RSA-AES128-SHA256",
                             "DHE-DSS-AES128-GCM-SHA256",
                             "DHE-DSS-AES128-SHA256","AES128-GCM-SHA256",
                             "AES128-SHA256","ECDHE-ECDSA-AES256-SHA",
                             "ECDHE-RSA-AES256-SHA","DHE-DSS-AES256-SHA",
                             "ECDH-ECDSA-AES256-SHA","ECDH-RSA-AES256-SHA",
                             "AES256-SHA","ECDHE-ECDSA-AES128-SHA",
                             "ECDHE-RSA-AES128-SHA","DHE-DSS-AES128-SHA",
                             "ECDH-ECDSA-AES128-SHA","ECDH-RSA-AES128-SHA",
                             "AES128-SHA"]},
                   {certfile,"etc/certs/client-cert.pem"},
                   {cacertfile,"etc/certs/cacert.pem"}]},
        {ssl,false},
        {retry_interval,20000},
        {proto_ver,v4},
        {password,'***'},
        {max_inflight_size,32},
        {keepalive,60},
        {forward_mountpoint,"/"},
        {clientid,"bridge_aws"},
        {clean_start,false},
        {bridge_mode,true},
        {batch_size,32},
        {address,"xx.xxx.xxx.xxx:1883"}]
reason:{error,ack_timeout}
(7ef2c6278889@172.17.0.2)1> 2020-07-28 12:38:56.484 [error] [Bridge] Can't be found from the inflight:3
(7ef2c6278889@172.17.0.2)1> 2020-07-28 12:38:56.875 [error] [Bridge] Can't be found from the inflight:4
(7ef2c6278889@172.17.0.2)1> 2020-07-28 12:38:57.130 [error] [Bridge] Can't be found from the inflight:5
(7ef2c6278889@172.17.0.2)1> 2020-07-28 12:38:57.384 [error] [Bridge] Can't be found from the inflight:6
(7ef2c6278889@172.17.0.2)1> 2020-07-28 12:38:57.553 [error] [Bridge] Can't be found from the inflight:7
(7ef2c6278889@172.17.0.2)1> 2020-07-28 12:38:58.032 [error] [Bridge] Can't be found from the inflight:8
(7ef2c6278889@172.17.0.2)1> 2020-07-28 12:38:58.286 [error] [Bridge] Can't be found from the inflight:9
...
...
...
```

Look forward to your answers.
